state/spacesdiscovery.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/utils/set"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/instance"
+	"github.com/juju/juju/network"
+)
+
+func (st *State) getModelSubnets() (set.Strings, error) {
```

Comment:
I'll leave it as it is for now, the 2nd part of this work (post-2.2) will be blocking add-{subnet,space}, support renaming, etc.